Add Jira support

ricky-rav opened the pull request:
Refactored code and added support for jira queries (bugs and escalations)

Signed-off-by: Riccardo Ravaioli <rravaiol@redhat.com>

Commit:
Add Jira support

Refactored code and added support for jira queries (bugs and escalations)

Signed-off-by: Riccardo Ravaioli <rravaiol@redhat.com>

ricky-rav commented:
/assign @flavio-fernandes

flavio-fernandes commented on README.md:
````diff
@@ -1,114 +1,90 @@
+
+## Download source code
+```
+$ git clone https://github.com/dougsland/bz-query && pushd bz-query
````
@ricky-rav I think we need to refine the install a little more.
I tried followoing the steps described here and end up with this error:
```
env ❯ ./network_bugs_overview -h
Traceback (most recent call last):
  File "/home/vagrant/dev/douglas_bugzilla/tmp/bz-query/./network_bugs_overview", line 8, in <module>
    from dateutil import parser
ModuleNotFoundError: No module named 'dateutil'
```

How about we create a file called requirements.txt with the dependencies we need:

```
jira
python-dateutil
```

and the as part of the install add `pip install .`

flavio-fernandes reviewed (changes requested):
A few more tweaks to the readme install, please.

flavio-fernandes commented on README.md:
```diff
@@ -1,114 +1,95 @@
-**ATENTION:**  
-As soon you click: "**Generate the API-key**" a long string will be generated with chars and numbers, COPY as it's only displayed **ONCE**.
+In particular, it queries:
+- bugzilla for existing bugs (the creation new bugs will soon be disabled on our bugzilla server), 
```
nit: white space in the end of this line

flavio-fernandes reviewed (changes requested):
hm.... I'm getting this error below. Maybe I'm doing something wrong?

.env ❯ git rev-parse HEAD
eb958cf203c11c4c9209bf9592f0b73d3249cc4a

Also, can you please format the python code with black:
```
pip install black
black network_bugs_overview
```
?

```
.env ❯ ./network_bugs_overview  --jira-bugs
Error running JIRA query project=OCPBUGS and component in ("networking/openshift-sdn", "networking/ovn-kubernetes", "networking") and status in ("NEW", "ASSIGNED", "POST", "ON_DEV", "In Progress"), error: Unterminated string starting at: line 1 column 145102 (char 145101)
- Found 0 bugs with the query: project=OCPBUGS and component in ("networking/openshift-sdn", "networking/ovn-kubernetes", "networking") and status in ("NEW", "ASSIGNED", "POST", "ON_DEV", "In Progress")
  Query processing time: 0.8s
  ```

dougsland commented:
@ricky-rav @flavio-fernandes awesome patch. +1 for merge it for now. However, as it's getting `pro-level` :)  Should we move to an `official team repo` somewhere?

ricky-rav commented:
> @ricky-rav @flavio-fernandes awesome patch. +1 for merge it for now. However, as it's getting `pro-level` :) Should we move to an `official team repo` somewhere?

no rush for now, I think :)

ricky-rav commented:
@flavio-fernandes I ran black on the source file and applied your requested changes.

Also, after clarifying this with the team yesterday, I removed from the default behaviour the queries to the RHOCPPRIO project in Jira. We'll see, as we get more familiar with jira bugs, what escalated bugs in OCPBUGS project will look like, how to include them, and whether we will use RHOCPPRIO issues at all.

Commit:
Fix component names, run jira queries until success and print summary table

- fixed component names
- the query to jira server is now run in a loop 100 times until we get a response or fail violently  (the server is very unreliable)
- print by default a summary table, to show more details about the bug load of each developer
- print with "-v" the full output we're used to
- remove "in progress" status for jira bugs, since it will be soon removed from the server

Signed-off-by: Riccardo Ravaioli <rravaiol@redhat.com>

dougsland commented:
looks good :) @flavio-fernandes ?

flavio-fernandes commented:
/lgtm!
